app/qml/editor/inputrelation.qml

```diff
@@ -0,0 +1,268 @@
+  }
+
+  anchors {
+    left: parent.left
+    right: parent.right
+  }
+
+  height: content.height
+
+  Item {
+    id: content
+
+    height: mode === "text" ? textModeContainer.height : photoModeContainer.height
+    anchors {
+      left: parent.left
+      right: parent.right
+    }
+
+    // Text Mode Widget
+    Rectangle {
+      id: textModeContainer
+
+      states: [
+        State {
+          name: "initial"
+          PropertyChanges {
+            target: noOfFeaturesText
+            visible: true
+          }
+          PropertyChanges {
+            target: textModeContainer
+            height: customStyle.fields.height
+          }
+        },
+        State {
+          name: "expanded"
+          PropertyChanges {
+            target: textModeContainer
+            height: customStyle.fields.height * 3 // three rows
+          }
+          PropertyChanges {
+            target: noOfFeaturesText
+            visible: false
+          }
+        },
+        State {
+          name: "page"
+          // TODO: page state
+        }
+      ]
+
+      transitions: [
+        Transition {
+          from: "initial"
+          to: "expanded"
+          animations: NumberAnimation { property: "height"; duration: 100 }
+        }
+      ]
+
+      onStateChanged: delegateModel.update()
+
+      visible: mode === "text"
+      height: customStyle.fields.height
+      width: parent.width
+      state: "initial"
+
+      border.color: customStyle.fields.normalColor
+      border.width: 1 * QgsQuick.Utils.dp
+      color: customStyle.fields.backgroundColor
+      radius: customStyle.fields.cornerRadius
+
+      Text {
+        id: noOfFeaturesText
+
+        anchors.fill: parent
+        font.pointSize: customStyle.fields.fontPointSize
+        color: customStyle.fields.fontColor
+        horizontalAlignment: Qt.AlignHCenter
+        verticalAlignment: Qt.AlignVCenter
+
+        text: qsTr( "%n linked feature(s)", "Shows how many features are linked via relations", root.linkedFeaturesCount )
+      }
+
+      MouseArea {
+        anchors.fill: parent
+        onClicked: {
+          if ( textModeContainer.state === "initial" )
+            textModeContainer.state = "expanded"
+          else if ( textModeContainer.state === "expanded" && root.canOpenFeaturesPage )
+            textModeContainer.state = "page"
+        }
+      }
+
+      Flow {
+        id: flowItemView
+
+        anchors.fill: parent
+        anchors.margins: customStyle.fields.sideMargin
+
+        spacing: customStyle.group.spacing
+
+        Repeater { model: delegateModel; }
+      }
+    }
+
+    // Photo Mode Widget
+    Item {
+      id: photoModeContainer
+
+      visible: mode === "photo"
+      anchors.fill: parent
+
+      // todo: photo panel widget
+    }
+  }
+
+  Component {
+    id: textDelegate
+
+    Item {
+      id: textDelegateContainer
+
+      property real maximumWidth: flowItemView.width
+      property bool isVisible: {
```

Comment:
It is necessary to limit elements, however some indicator (e.g three dots) is missing in such case

Reply:
Yes, I plan to add it in next PR. It is indeed in the docs.